File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class AnalyticsSummaryResponse(BaseModel):
    total_calls: int
    avg_duration: float
    escalation_rate: float
    sentiment_avg: float | None


class CallVolumePoint(BaseModel):
    date: str
    count: int


class EscalationReasonPoint(BaseModel):
    reason: str
    count: int


class DurationByDayPoint(BaseModel):
    day: str
    avg_duration: float


class AnalyticsSeriesResponse(BaseModel):
    call_volume: list[CallVolumePoint]
    escalation_reasons: list[EscalationReasonPoint]
    duration_by_day: list[DurationByDayPoint]

File: frontend/src/api/analytics.ts
import { apiClient } from "./client";
import type { AnalyticsSummary } from "@/types";

export type AnalyticsSeries = {
  call_volume: { date: string; count: number }[];
  escalation_reasons: { reason: string; count: number }[];
  duration_by_day: { day: string; avg_duration: number }[];
};

export async function getAnalyticsSummary(
  businessId: string,
  period: string
): Promise<AnalyticsSummary> {
  const response = await apiClient.get<AnalyticsSummary>("/analytics/summary", {
    params: { business_id: businessId, period },
  });
  return response.data;
}

export async function getAnalyticsSeries(
  businessId: string,
  period: string
): Promise<AnalyticsSeries> {
  const response = await apiClient.get<AnalyticsSeries>("/analytics/series", {
    params: { business_id: businessId, period },
  });
  return response.data;
}
